src/semantic_hashing_demo/lsh.py

```diff
@@ -0,0 +1,156 @@
+#     # query = 'Excellent value for delicious taffy. The selection offered a broad variety of delectable flavors. The shipping was impressively fast. For enthusiasts of taffy, this offer is a must-grab.'    # ai-generated the 5th review a bit
+#     # query = "This dog food is highly nutritious and beneficial for digestive health. It's also suitable for young puppies. My dog consistently consumes the recommended portion at each meal."  # ai-generated the index-9 review a bit
+
+#     # ===== Type-4 =====
+#     # query = "I have lived out of the US for over 7 yrs now, and I so miss my Twizzlers!!  When I go back to visit or someone visits me, I always stock up.  All I can say is YUM!<br />Sell these in Mexico and you will have a faithful buyer, more often than I'm able to buy them right now."
+#     # query = "Good oatmeal.  I like the apple cinnamon the best.  Though I wouldn't follow the directions on the package since it always comes out too soupy for my taste.  That could just be me since I like my oatmeal really thick to add some milk on top of."
+#     # query = "I roast at home with a stove-top popcorn popper (but I do it outside, of course). These beans (Coffee Bean Direct Green Mexican Altura) seem to be well-suited for this method. The first and second cracks are distinct, and I've roasted the beans from medium to slightly dark with great results every time. The aroma is strong and persistent. The taste is smooth, velvety, yet lively."
+#     hash_query = hash_vector(get_embedding(query), plane_norms)
+#     # hash_query = hash_vector(get_embedding(query, model), plane_norms)
+#     print(f"\nFor a given text: \"{query}\", it's computed hash is '{hash_query}'.")
+
+#     # calculate the hamming distance between the query and each bucket
+#     print("\nhamming distances b/w the query from each bucket key:")
+#     hamming_distances = []
+#     for hash_str in buckets.keys():
+#         hamming_distances.append(hamming_distance(hash_query, hash_str))
+#     print(hamming_distances)
+#     if contains_zero(hamming_distances):
+#         print(
+#             "🙂 The given text falls into the bucket with its key having exact same hash"
+#         )
+#     else:
+#         print(
+#             "☹️ As no exact hash found, deliberately the closest bucket with min. hamming distance is selected here from left --> right."
+#         )
+#     # Get the index of the lowest one
+#     min_index = np.argmin(hamming_distances)
+#     print(
+#         f"\nHence, the given text belongs to the index-{min_index} of bucket list, \ni.e. the bucket with key: '{list(buckets.keys())[min_index]}', value: [{list(buckets.values())[min_index]}]."
+#     )
+
+
+# if __name__ == "__main__":
+#     main()
```

Comment:
This can be deleted also.

Reply:
[Commit](https://github.com/abhi3700/semantic-hashing-demo/pull/2/commits/506a8c9ea571e9706b10a1b80b2ac0ff8959dd61).